tests/modules/asset_groups/resources/test_create_asset_group.py

```diff
@@ -288,6 +288,78 @@ def test_create_asset_group_detection(
         tus_utils.cleanup_tus_dir(transaction_id)
 
 
+def test_create_asset_group_repeat_detection(
+    flask_app,
+    flask_app_client,
+    researcher_1,
+    internal_user,
+    staff_user,
+    test_root,
+    db,
+    request,
+):
+
+    (
+        transaction_id,
+        asset_group_uuid,
+        asset_group_sighting_uuid,
+    ) = asset_group_utils.create_asset_group_to_curation(
+        flask_app, flask_app_client, researcher_1, internal_user, test_root
+    )
+    request.addfinalizer(lambda: tus_utils.cleanup_tus_dir(transaction_id))
+    request.addfinalizer(
+        lambda: asset_group_utils.delete_asset_group(
+            flask_app_client, researcher_1, asset_group_uuid
+        )
+    )
+
+    # Rotate one of the assets, No API for this apparently?
+    from app.modules.asset_groups.models import (
+        AssetGroup,
+        AssetGroupSighting,
+        AssetGroupSightingStage,
+    )
+
+    asset_group = AssetGroup.query.get(asset_group_uuid)
+    asset_group.assets[0].rotate(45)
+
+    # Simulate a valid response from Sage but don't actually send the request to Sage
+    with mock.patch.object(
+        flask_app.acm,
+        'request_passthrough_result',
+        return_value={'success': True},
+    ) as detection_reran:
+        asset_group_utils.detect_asset_group_sighting(
```

Comment:
In order to assert something that a background task has done, we need to change the background task to run in the foreground:

```python
    from app.modules.asset_groups import tasks

    with mock.patch.object(
        tasks.sage_detection,
        'delay',
        side_effect=lambda *args, **kwargs: tasks.sage_detection(*args, **kwargs),
    ):
        asset_group_utils.detect_asset_group_sighting(
```

Here we're changing `.delay()` to just call the task code.

Reply:
Thanks, That unblocked that strangeness.